qbraid_qir/qasm3/convert.py

```diff
@@ -69,14 +81,23 @@ def qasm3_to_qir(
 
     final_module = QasmQIRModule(name, qasm3_module, llvm_module)
 
-    profile = profile.lower()
+    # Convert enum to string if needed, then normalize and validate
+    if isinstance(profile, Profile):
+        profile_str = profile.value.lower()
+    else:
+        profile_str = profile.lower()
+        # Validate that the string corresponds to a valid profile
+        valid_profiles = [p.value for p in Profile]
+        if profile_str not in valid_profiles:
+            raise ValueError(f"Invalid profile: {profile}. Valid profiles are: {valid_profiles}")
+
     visitor: Any
-    if profile == "adaptive":
+    if profile_str == "adaptive":
         visitor = QasmQIRAdaptiveVisitor(external_gates=external_gates, **kwargs)
-    elif profile == "base":
+    elif profile_str == "base":
         visitor = QasmQIRVisitor(external_gates=external_gates, **kwargs)
     else:
-        raise ValueError(f"Invalid profile: {profile}")
+        raise ValueError(f"Invalid profile: {profile_str}")
```

Comment:
Should be a `NotImplementedError` as the profile will at least be valid but we might not have an associated visitor (or handler) for it